In [1]:
from pathlib import Path

data_root = Path("../data/raw")

for split in ["train", "test"]:
    for cls in ["def_front", "ok_front"]:
        path = data_root / split / cls
        count = len(list(path.glob("*")))
        print(f"{split}/{cls}: {count}")

train/def_front: 3758
train/ok_front: 2875
test/def_front: 453
test/ok_front: 262


## Findings

### 1. Class balance
- Train: def_front=3758, ok_front=2875 (~57/43 split)
- Test: def_front=453, ok_front=262 (~63/37 split)
- Moderate imbalance, not severe. Decisions:
  - Use per-class precision/recall/F1 (or balanced accuracy) as the primary metric, not raw accuracy — a majority-class classifier already scores ~57-63%.
  - Consider class-weighted loss in training (Stage 3).
- Train/test imbalance ratios differ slightly — test isn't a perfectly stratified subset of train. Worth keeping in mind, not a blocker.